### PR TITLE
Fix "500 internal error" when gpfdist fails to receive data from network(6X_STABLE).

### PR DESCRIPTION
**Pull request by zhaorui20005**
When insert into writable external tables, the gpfdist is the server, and segments are clients. The clients will send POST requests one by one in sequence order. There is retry mechanism on the gpdb segment side when it fails to send one POST request to gpfdist because of network failure. If the network failure happens after the connection has been made, then gpfdist will also get an error when reading data from the socket. 
However, the retry is not working because of gpfdist legacy code that will close fstream when failed to receive data from socket. Gpfdist will close the file stream and set an error, so when the retry POST request arrives to gpfdist, an "500 internal error" will be sent back to segment which will stop the whole insert procedure.

This commit will let gpfdist confirm the receiving data from of the POST request without error and then write into the local file stream. After sending "HTTP/1.0 200 ok" successfully, gpfdist will add the SEQ number. It will not release the file stream because of network error. And when gpfdist receives the retry POST request again, it continues to process until successfully. So this could allow segment retry sending POST requests to gpfdist to avoid 500 error because of network failure.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by hyongtao-db — APPROVED** [review]
LTGM. Just some tiny typos and format problems. I have helped you rewrite them.

**Comment by zhaorui20005** [comment]
> LTGM. Just some tiny typos and format problems. I have helped you rewrite them.

Thanks for the fixing.

File: src/bin/gpfdist/gpfdist.c [hunk 1]
```diff
@@ -3069,11 +3068,31 @@ static void handle_post_request(request_t *r, int header_end)
...
+		if (write_file_buffer == NULL) 
+		{
+			gfatal(r, "Cannot alloc memory for write_file_buffer");
+		}
+	}
+	if (r->in.davailable > (int) write_file_size) 
+	{
+		while(r->in.davailable > (int) write_file_size) 
+		{
+			write_file_size <<= 1;
```
Comment (huiliang-liu) [inline src/bin/gpfdist/gpfdist.c:3083]: do we need to limit max size to avoid a bad request making memory out of control?

**Review by huiliang-liu — APPROVED** [review]
Thanks for your contribution! 
It is better to mention that the 500 error is hard to reproduce on CI, and it has been tested manually

File: src/bin/gpfdist/gpfdist.c (hunk 1 above)
Comment (zhaorui20005) [reply src/bin/gpfdist/gpfdist.c:3101]: Yeah, I will add one size range check for this.

**Pull request merged**